nextflow/modules/mashmap.nf

```diff
@@ -0,0 +1,19 @@
+process mashmap {
+      label 'mashmap'
+
+      publishDir "${params.output}/${assembly_name}/", mode: 'copy', pattern: "*.tsv"
+      publishDir "${params.output}/${assembly_name}/${params.finaldir}/mashmap", mode: 'copy', pattern: "*.tsv"
+
+    input:
+      tuple val(assembly_name), val(confidence_set_name), file(fasta) 
+      file(reference)
+      val(length) 
+    
+    output:
+      file("${confidence_set_name}_mashmap_hits.tsv")
+    
+    shell:
+    """
+    mashmap -q ${fasta} -r ${reference} -t ${params.cores} -o ${confidence_set_name}_mashmap_hits.tsv --noSplit -s ${length}
```

Comment:
* you can directly use `params.mashmap_len` here, since nf v20 params are global and can be simply used in modules

hint:
* the implementation works for input that should be compared to the same `reference` (and most likely this will be the case in one run) but maybe for a future extension and high-throughput processing let's assume you have two assemblies A and B and you want to mashmap them against reference refA and refB  

Now it's unclear for nextflow which reference belongs to which assembly. This can be resolved by the `--list` input:

```bash
#assemblies.csv
A,a.fasta
B,b.fasta

#mashmap.csv
A,refA.fasta
B,refB.fasta
```

And now instead of 

```java
            mashmap(predicted_contigs, mashmap_ref_ch)
```

one can do

```java
            mashmap(predicted_contigs.join(mashmap_ref_ch, by: 0))
```

and some other slight changes would need to be introduced. But by doing so, the correct references could be assigned to the correct assemblies. 

Totally fine like it is implemented now, I just wanted to mention this :)

Reply:
ah and use `task.cpus` instead of `params.cores`. 

`params.cores` refers to the parameter in the `nextflow.config` file. 

but let's say you run the pipeline in the cloud where you defined in the corresponding config `cores '20'` for mashmap, this will not be used because you access this value via `task.cpus`. 

that's why you wrote 

```
withLabel: mashmap {      cpus = params.cores;   container = 'quay.io/biocontainers/mashmap:2.0--gsl2.2_1' }
```

`nextflow run virify.nf --cores 4` will set `params.cores = 4` and then `task.cpus = 4`